apps/api/src/app/sessions/audio-session.interceptor.ts

```diff
@@ -0,0 +1,25 @@
+import { CallHandler, ExecutionContext, Injectable, NestInterceptor } from '@nestjs/common';
+import { Observable, map } from 'rxjs';
+import { Session } from './entities/session.entity';
+import { isArray } from 'class-validator';
+
+@Injectable()
+export class AudioSessionInterceptor implements NestInterceptor {
+  intercept(context: ExecutionContext, next: CallHandler): Observable<Session | Session[]> {
+    return next.handle().pipe(
+      map((data: any) => {
+        if (isArray(data)) {
```

Comment:
is this the same as Array.isArray function?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray

Reply:
Not sure what is used inside class-validator for the array check. In the end I used it under the assumption that we also trust class-validator with this in input objects/DTOs, so should be good enough here as well (so far it differentiated correctly)